src/Swarm/Game/Step.hs

```diff
@@ -1065,6 +1065,161 @@ seedProgram minTime randTime thing =
     selfdestruct
   |]
 
+-- Triggers the ignition of the entity underfoot with some delay.
```

Comment:
All the code for dealing with ignition & combustion is somewhat complex.  I think I understand it and it makes sense.  But it might be nice to have a comment somewhere giving an overall explanation of the flow of execution and how all the pieces fit together.

We could also move all the combustion-related code into a submodule, `Swarm.Game.Step.Combustion`?  (The `Step` module is long enough as it is!)  That would also provide a natural place for a comment explaining the big picture.

Reply:
I moved just enough helper functions out of `Swarm.Game.Step` and into `Swarm.Game.Step.Util` to allow for extraction of the combustion code into `Swarm.Game.Step.Combustion`.  We can follow up to offload more stuff to `Swarm.Game.Step.Util` and re-organize.